crates/nu-engine/src/maybe_text_codec.rs

```diff
@@ -1,113 +1,65 @@
-use std::io::{BufRead, BufReader, Read};
+use std::io::{BufRead, BufReader, Error, Read};
 
-use nu_errors::ShellError;
-
-use encoding_rs::{CoderResult, Decoder, Encoding, UTF_8};
-
-#[cfg(not(test))]
-const OUTPUT_BUFFER_SIZE: usize = 8192;
-#[cfg(test)]
-const OUTPUT_BUFFER_SIZE: usize = 4;
```

Comment:
We can have simplified cases for when we know that the input is finite, but for the general case, we can't get rid of buffers and windows of input.